model/src/main/kotlin/utils/OrtResultExtensions.kt

```diff
@@ -84,3 +90,36 @@ fun OrtResult.getDetectedLicensesWithCopyrights(
     omitExcluded: Boolean = true
 ): Map<String, Set<String>> =
     LicenseResolver(this, packageConfigurationProvider).getDetectedLicensesWithCopyrights(id, omitExcluded)
+
+/**
+ * Create [PackageConfiguration]s for the projects in this [OrtResult]. [PathExclude]s and [LicenseFindingCuration]s are
+ * taken from the [RepositoryConfiguration]. The [Provenance] is taken from the scan results. Projects without scan
+ * result and scan results without provenance information are ignored.
+ */
+fun OrtResult.getProjectConfigurations(): List<PackageConfiguration> {
+    fun VcsInfo.toMatcher() = VcsMatcher(
+        type = type,
+        url = url,
+        revision = resolvedRevision ?: revision,
+        path = path
+    )
+
+    return analyzer?.result?.projects.orEmpty().flatMap { project ->
+        getScanResultsForId(project.id).mapNotNull { scanResult ->
+            val sourceArtifactUrl = scanResult.provenance.sourceArtifact?.url
```

Comment:
I believe the source artifact should always be null as for projects we only ever scan the VCS.

Reply:
I didn't want to make this assumption here, in case this changes sometime.

Reply:
How about the argumentation then?

A set of {path excludes, license finding curations } can be applied only to exactly one of the following
- source artifact
- VCS / fixed revision

..but never to both, because the file contents may differ between VCS and source artifact.
Applying it to both would thus make it harder and more complex to figure out whether a specific entry is needed and to what it applies to. Further it would not allow a workflow for support which sets up configuration scan by scan, because the configuration is mixed. Developing automation / tooling on top of PackageConfigurations would be made harder because of this mixing of configurations.

Finally, the curations and path excludes for projects are defined in the `RepositoryConfiguration`, and the documentation of a RepositoryConfiguration.exlucdes` says: "Defines which parts of the repository will be excluded...".

I think not separating things introduces too many potential pitfalls (without need) and should thus be avoided.

Reply:
I ignore the source artifact now and added a code comment explaining why.